apm-agent-core/src/test/java/co/elastic/apm/agent/MockReporter.java

```diff
@@ -592,21 +590,34 @@ public void awaitUntilAsserted(ThrowingRunnable runnable) {
      * This is an issue when testing instrumentations that instrument {@link java.util.concurrent.Executor}.
      *
      * @param timeoutMs the timeout of the condition
-     * @param runnable  a runnable that trows an exception if the condition is not met
+     * @param runnable  a runnable that throws an exception if the condition is not met
      */
     public void awaitUntilAsserted(long timeoutMs, ThrowingRunnable runnable) {
         Throwable thrown = null;
         for (int i = 0; i < timeoutMs; i += 5) {
             try {
                 runnable.run();
-                thrown = null;
+                return;
             } catch (Throwable e) {
                 thrown = e;
                 LockSupport.parkNanos(TimeUnit.MILLISECONDS.toNanos(5));
             }
         }
-        if (thrown != null) {
-            throw new RuntimeException(String.format("Condition not fulfilled within %d ms", timeoutMs), thrown);
+        throw new RuntimeException(String.format("Condition not fulfilled within %d ms", timeoutMs), thrown);
+    }
+
+    /**
+     * @param timeoutMs timeout of the condition
+     * @param runnable  a runnable that throws an exception when the condition is not met
+     */
+    public void awaitUntilTimeout(long timeoutMs, ThrowingRunnable runnable) {
```

Comment:
Isn't that effectively the same as
```
Thread.sleep(timeoutMs);
runnable.run();
```

Reply:
That's quite close, with the minor difference that it's failing faster when the assertion throws.

Reply:
ah, right